packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,193 @@
+import * as ts from 'typescript';
+import { excludeDeclarationFiles, ConfigurableRule, Replacement } from '@fimbul/ymir';
+import {
+    getLineBreakStyle,
+    isCallExpression,
+    isExpressionStatement,
+    isClassDeclaration,
+    isClassExpression,
+    isConstructorDeclaration,
+    isParameterProperty,
+    isBinaryExpression,
+    isPropertyAccessExpression,
+    isIdentifier,
+    isPropertyDeclaration,
+    hasSideEffects,
+} from 'tsutils';
+
+export interface Options {
+    mode: 'when-possible' | 'never' | 'consistent';
+}
+
+@excludeDeclarationFiles
```

Comment:
also add `@typescriptOnly` as JavaScript files cannot have parameter properties. consider adding a test for that